cli/internal/fs/copy_file.go

```diff
@@ -61,10 +65,10 @@ func RecursiveCopyOrLinkFile(from string, to string, mode os.FileMode, link, fal
 			} else if isSame {
 				return nil
 			}
-			return CopyOrLinkFile(name, dest, fileMode, mode, link, fallback)
```

Comment:
This `fileMode`, passed to the `WalkMode` callback is what I found to be incorrect (always 0)

Reply:
So, this is "correct", but it's not what anybody might expect:

https://github.com/karrick/godirwalk/blob/9a7752c108e7ea76255201b9f47bd4d4d2df868e/modeType.go#L19

It's explicitly *not* a complete `FileMode`, it's masked to only include only `os.ModeType`.

Anyway, I didn't go looking for why that is until you made this note, this is definitely an API footgun and I believe it's likely why we had this bug to begin with.

Reply:
Ah interesting, yea this wasn't what I was expecting at all. Thanks for digging into this.